src/main/java/frc/robot/commands/AutoCenter.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import com.ctre.phoenix6.mechanisms.swerve.SwerveRequest;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.LimelightHelpers;
+import frc.robot.RobotContainer;
+import frc.robot.subsystems.SwerveSubsystem;
+
+public class AutoCenter extends Command {
+  
+  boolean end = false;
+  double kP = -0.05;
+  private final Timer timer = new Timer();
+  private final SwerveSubsystem driveSubsystem;
+  private final SwerveRequest.FieldCentric drive;
+
+  private final ShootCommand shootCommand = new ShootCommand(RobotContainer.shooterSubsystem, RobotContainer.intakeSubsystem);
+
+  private final PrepCommand prep = new PrepCommand(RobotContainer.shooterSubsystem, 45, 50, true);
+  
+  /** Creates a new AutoCenter. */
+  public AutoCenter(SwerveSubsystem driveSubsystem, SwerveRequest.FieldCentric drive) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    this.driveSubsystem = driveSubsystem;
+    this.drive = drive;
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    
+    timer.start();
+    
+    end = false;
+    
+    // if the limelight doesn't see an april tag, stop the command
+    if (LimelightHelpers.getTX("") == 0) {
+        end = true; 
```

Comment:
fwiw, you should also be able to run `this.cancel()` and not have to worry about isFinished being true or false. It will have the scheduler call `end(true)` and you can use the `interrupted` variable there to determine if you should shoot or not.